INSTALL

```diff
@@ -5,54 +5,117 @@ Subsurface uses quite a few open source libraries and frameworks to do its
+believe developers should build against. All our patches are contained
+in the "Subsurface-branch" which are also HEAD.
+
+This should allow distros to see which patches we have applied on top of
 upstream. They will receive force pushes as we rebase to newer versions of
 upstream so they are not ideal for ongoing development (but they are of
 course easy to use for distributions as they always build "from scratch",
 anyway).
 
-The rationale for this is that we have no intention of forking either of
-these two projects. We simply are adding a few patches on top of their
-latest versions and want to do so in a manner that is both easy for our
+The rationale for this is that we have no intention of forking the
+project. We simply are adding a few patches on top of their latest
+version and want to do so in a manner that is both easy for our
 developers who try to keep them updated frequently, and anyone packaging
 Subsurface or trying to understand what we have done relative to their
 respective upstreams.
 
-At this point Qt5 is required, Qt5.4 or newer is recommended and on the
-Mac, in order to get native Bluetooth support, Qt5.5 is necessary.
 
-Similarly, in order for our cloud storage to be fully functional you need
+Getting Qt5
+-----------
+
+We use Qt5 in order to only maintain one UI across platforms. Qt5.8 is the
+oldest version supported.
+
+Download the Open Source version from https://www.qt.io/download and 
+follow the Qt instructions to install it or alternatively follow the
+instruction specific to a distribution (see build instructions).
+
+To save time and disk space you can unselect Android and IOS packages
+(Of course unless you want to build Android/iOS versions) as well as
+- Qt Data Visualisation
+- Qt Purchasing
+- Qt Virtual Keyboard",
+- Qt WebEngine
+- Qt Network Authorization
+- Qt Remote Objects
+- Qt WebGL Streaming
+- Qt Script
+- Qt 3D2
+- Qt Canvas 3D
+- Qt Extras
+This can be done later by use of MaintenanceTool.app in your Qt folder.
+
+Remark: most Qt installations does not make Qt available on the command
+line, which is needed. In order to make Qt available you need to 
+add a symbol link (do something like
+ln -s <Qt location>/<version>/<type>/bin/qmake /usr/local/bin/qmake
+
+QtWebKit is needed, if you want to print, but no longer part of Qt5,
+so you need to download it and compile. In case you just want to test
+without print possibility omit this step.
+
+
+Other third party library dependencies
+--------------------------------------
+
+In order for our cloud storage to be fully functional you need
 libgit2 0.23 or newer.
 
-Finally, as of Subsurface 4.5 we have switched our build system to cmake.
-qmake based builds are no longer supported.
+
+cmake build system
+------------------
+
+As of Subsurface 4.5 we have switched our build system to cmake.
+qmake based builds are no longer supported. Remark even though
+no longer supported, qmake is still used by googlemaps and the iOS build.
```

Comment:
Historically this made sense. I think I'd now would phrase this differently (for people like you, coming new to the project):

Our main build system is based on cmake. But qmake is needed for the googlemaps
plugin and the iOS build.

Reply:
Done, and pushed a new commit. I will try to work with --amend in the future, different projects different styles :-)

Reply:
It's a fine line. And I know that some projects have very different views on this.
I want people to do multiple commits when making multiple changes. And I want these commits to stay in our repo - so all these projects that by default always squish together every PR... they are IMHO doing it wrong.
But at the same time, as you update things based on review feedback, those changes should be merged together.
So in the case that you have a PR with four commits, sometimes we'll ask you to update a commit in the middle and then do a force push so we have a clean, useful git history.
Makes sense?

Reply:
It makes a lot of sense, I am very careful to only have 1 change/problem in 1 commit, but of course the pull request, might contain a couple of those. For this document change it makes sense to squash them all.

Actually I am preparing some changes to build.sh just like that:
1) add some "rm" to do cleanup
2) add my new build option -prereq, that will check for missing tools